docs/reference/scripting.asciidoc

```diff
@@ -4,61 +4,76 @@
 [partintro]
 --
 With scripting, you can evaluate custom expressions in {es}. For example, you
-could use a script to return "script fields" as part of a search request or
+could use a script to return fields as part of a search request or
 evaluate a custom score for a query.
 
-The default scripting language is <<modules-scripting-painless, `Painless`>>.
+The default scripting language is <<modules-scripting-painless,Painless>>.
 Additional `lang` plugins enable you to run scripts written in other languages.
 Everywhere a script can be used, you can include a `lang` parameter
 to specify the language of the script.
 
 [discrete]
-== General-purpose languages
+[[scripts-and-search-speed]]
+== Scripts and search speed
 
-These languages can be used for any purpose in the scripting APIs,
-and give the most flexibility.
+Scripts can't make use of {es}'s index structures or related optimizations,
+which can sometimes result in slower search speeds.
 
-[cols="<,<,<",options="header",]
-|=======================================================================
-|Language
-    |Sandboxed
-    |Required plugin
+If you often use scripts to transform indexed data, you can speed up search by
+making these changes during ingest instead. However, that often means slower
+index speeds.
 
-|<<modules-scripting-painless, `painless`>>
-    |yes
-    |built-in
+[discrete]
+[[modules-scripting-using-caching]]
+== Script caching
+
+All scripts are cached by default so that they only need to be recompiled
```

Comment:
The format of these settings has changed, as we want to move back to the format described here, it's fine to keep these (they still work right now).

_however_, if we refer to this format, it does not make sense to include the line.
>For most contexts, the default cache size is `100`. For ingest contexts, the
default cache size is `200`.

Reply:
++
I'll remove that line.